### PR TITLE
Update all hooks to use advanced results

### PR DESCRIPTION
**Pull request by phobologic**
This takes advantage of GH-270 for all hooks now.

### CODE REVIEW

**Comment by phobologic** [comment]
@danielkza - since you're probably the most familiar with this feature, any chance you could take a look?

File: stacker/hooks/ecs.py [hunk 1]
```diff
@@ -29,4 +29,4 @@ def create_clusters(region, namespace, mappings, parameters, **kwargs):
     for cluster in clusters:
         logger.debug("Creating ECS cluster: %s", cluster)
         conn.create_cluster(clusterName=cluster)
-    return True
+    return {"clusters": clusters}
```
Comment (danielkza) [inline stacker/hooks/ecs.py]: Wouldn' it be more useful to return the results of the `create_cluster` calls?

File: stacker/hooks/iam.py [hunk 2]
```diff
@@ -139,10 +141,15 @@ def ensure_server_cert_exists(region, namespace, mappings, parameters,
...
+        result = "uploaded"
         logger.info(
             "uploaded certificate: %s (%s)",
             cert_name,
             cert_arn,
         )
 
-    return True
+    return {
+        "result": result,
```
Comment (danielkza) [inline stacker/hooks/iam.py]: What about using `status` instead of `result`? Just a nitpick.

File: stacker/hooks/keypair.py [hunk 3]
```diff
@@ -26,7 +26,11 @@ def ensure_keypair_exists(region, namespace, mappings, parameters, **kwargs):
                     keypair["KeyName"],
                     keypair["KeyFingerprint"],
                     "exists")
-        return True
+        return {
+            "result": "exists",
```
Comment (danielkza) [inline stacker/hooks/keypair.py]: What about using status instead of result? Just a nitpick.

File: stacker/hooks/keypair.py [hunk 4]
```diff
@@ -50,7 +54,12 @@ def ensure_keypair_exists(region, namespace, mappings, parameters, **kwargs):
                     keypair["KeyName"],
                     keypair["KeyFingerprint"],
                     "imported")
-        return True
+        return {
+            "result": "imported",
```
Comment (danielkza) [inline stacker/hooks/keypair.py]: What about using status instead of result? Just a nitpick.

File: stacker/hooks/keypair.py [hunk 5]
```diff
@@ -71,10 +80,15 @@ def ensure_keypair_exists(region, namespace, mappings, parameters, **kwargs):
...
                     "created")
-        f = open(os.path.join(full_path, file_name), "w")
-        f.write(keypair["KeyMaterial"])
-        f.close()
-        return True
+        with open(os.path.join(full_path, file_name), "w") as f:
+            f.write(keypair["KeyMaterial"])
+
+        return {
+            "result": "created",
```
Comment (danielkza) [inline stacker/hooks/keypair.py]: What about using status instead of result? Just a nitpick.

**Comment by phobologic** [comment]
@danielkza updated, thanks for the review!

**Pull request merged**